public/static/js/settings.js

```diff
@@ -80,10 +81,49 @@ mciModule.controller('SettingsCtrl', ['$scope', '$http', '$window', 'notificatio
       });
   }
 
-  $scope.updateUserSettings = function(new_tz, use_spruce_options) {
+  $scope.showAdditionalFields = function(){
+    var additionalFieldCount = 1;
+    var additionalField = document.getElementById("feedback-" + additionalFieldCount);
+
+    while (additionalField) {
+      if (use_spruce_options.patch_page) {
+        additionalField.style.display = "block";
+      } else {
+        additionalField.style.display = "none";
+      }
+      additionalFieldCount++;
+      additionalField = document.getElementById("feedback-" + additionalFieldCount);
+    }
+  }
+
+  function formatFeedback(spruce_feedback) {
+    var formattedFeedback = { type: "new_patches_page_feedback" };
+    var questionAnswerArray = [];
+    var allFields = ["information_score", "usability_score", "missing_things", "requested_changes"];
+    allFields.forEach( function(field) {
+        var prompt = document.getElementById(field + "_prompt").innerHTML;
```

Comment:
I think a better pattern would be to do the reverse - set each prompt somewhere in $scope, then bind the appropriate label to that field

Reply:
Fixed